test/hotspot/jtreg/gc/stress/systemgc/TestSystemGC.java

```diff
@@ -77,7 +77,7 @@ class LongLivedAllocationTask extends Exitable implements Runnable {
     @Override
     public void run() {
         while (!shouldExit()) {
-            String prefix = "long" + System.currentTimeMillis();
+            String prefix = "long" + (System.nanoTime() % 10000); // limit to 4 digits after changing from milliseconds to nanoseconds, else the key will use more memory
```

Comment:
Is this used to calculate elapsed time? If not, I think using `currentTimeMillis` is fine.

Reply:
It is not to calculate elapsed time. It is to create a unique(ish) key for a hash table. The new code does not guarantee a unique value, but then neither did the old.

Reply:
I could possibly add some characters to the key if that is what bothers you. I could of course also use `currentTimeMillis` as it is nothing wrong with it. But I would prefer removing `currentTimeMillis` in all gc testing.

Reply:
I will try to see if I can do it without using time at all.

Reply:
Not using time is probably the best. Not obvious if #digits here is critical. Maybe this is useful -- e.g. to generate fixed 6-digit int:

```java
Random rnd = new Random();
int n = 100_000 + rnd.nextInt(900_000);
```

Reply:
I reverted to the original implementation as I did not understand the idea behind it.